omnigibson/systems/micro_particle_system.py

```diff
@@ -1491,42 +1353,173 @@ def cm_customize_particle_material(cls):
                 customize_particle_material(mat=cls._material)
 
         # Add to any other params specified
-        kwargs["_register_system"] = cp_register_system
-        kwargs["particle_contact_offset"] = cp_particle_contact_offset
-        kwargs["particle_density"] = cp_particle_density
         kwargs["_material_mtl_name"] = cp_material_mtl_name
         kwargs["_customize_particle_material"] = cm_customize_particle_material
 
         # Create and return the class
-        return subclass_factory(name=f"{fluid_name}System", base_classes=FluidSystem, **kwargs)
+        return super().create(
+            name=name,
+            particle_contact_offset=particle_contact_offset,
+            particle_density=particle_density,
+            **kwargs,
+        )
 
+def customize_particle_material_factory(attr, value):
+    def func(mat):
+        setattr(mat, attr, np.array(value))
```

Comment:
should we generalize to also consider single scalars (e.g: floats), not just arrays?

Reply:
I think the value should be an array because it represents an RGB color? Down in the row if we want further customization, we can make this function more complex.

Reply:
right, but there's many other values that are not just RGB values, e.g.: specular, or map_influence, etc. values that are a range from 0 to 1.0

But for now I'm fine just leaving it as is and implementing more robust stuff in the future